### DISCUSSION TITLE
feature: can I add IsMnemonic validation

### DISCUSSION THREAD

**Discussion by Hathoriel** [typestack/class-validator#1786] [Feature Ideas]
### Description
I would like to add mnemonic validation

### Proposed solution
```ts
import { registerDecorator, ValidationOptions } from 'class-validator'
import BigNumber from 'bignumber.js'

export const IsMnemonic = (validationOptions?: ValidationOptions) =>
  function (object: Object, propertyName: string) {
    registerDecorator({
      name: 'isMnemonic',
      target: object.constructor,
      propertyName: propertyName,
      options: {
        message: 'Mnemonic must contain between 12 and 24 words. Remove white spaces at end and start from your mnemonic.',
        ...validationOptions,
      },
      validator: {
        validate(value: any) {
          const bigNumberValue = new BigNumber(value.split(' ').length)
          return bigNumberValue.lte(24) && bigNumberValue.gte(12)
        },
      },
    })
  }
```

**Comment by braaar** [comment]
Since this introduces an additional dependency, I would be wary about adding it in this project. Seems perfectly okay to use a custom validator for this.

**Reply by NoNameProvided** [reply]
We have [BigInt](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt) now in JS, so the dependency can be dropped probably.

**Comment by NoNameProvided** [comment]
What is the use general case for this? I don't see this landing in the core, because I think this is a special use-case and better suited to be implemented in user-land via a custom decorator.
